### PR TITLE
[INLONG-5796][SDK] Optimize fetcher builders

### LINKED ISSUE #5796: [Improve][SDK] Optimize fetcher builders
### Description

1. Unify both single and multiple topic fetcher into one Builder. Differ them by 
2. Add params "fetch key" to specify each fetcher
3. Optimize the kafka rebalance process

### InLong Component

InLong SDK

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
- Fixes #5796 

### Motivation

1. Unify both single and multiple topic fetcher into one Builder. Differ them by
2. Add params "fetch key" to specify each fetcher
3. Optimize the kafka rebalance process, sleep 1s each time until all inflight packets have been ack.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java [hunk 1]
```diff
@@ -209,69 +216,67 @@ public List<InLongTopic> getTopics() {
...
+            removeOffsets.clear();
+            commitOffsetMap.put(topicPartition, new OffsetAndMetadata(commitOffset));
         });
     }
 
     public class Fetcher implements Runnable {
 
+        private boolean subscribeNew() {
+            if (CollectionUtils.isEmpty(newTopics)) {
+                //LOGGER.info("new topics is empty or null");
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java:255]: Remove it.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java [hunk 2]
```diff
@@ -18,79 +18,151 @@
...
+
+    private void ackRemovedTopicPartitions(Collection<TopicPartition> revoked) {
+        LOGGER.info("ack revoked topic partitions");
+        prepareCommit();
+        consumer.commitSync(commitOffsetMap);
+        // remove revoked topic partitions
+        Set<TopicPartition> keySet = ackOffsetMap.keySet();
+        revoked.stream()
+                .filter(keySet::contains)
+                //.peek(tp -> LOGGER.info("remove tp {}", tp))
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java:130]: Remove it.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java [hunk 3]
```diff
@@ -18,79 +18,151 @@
...
-            LOGGER.info("clusterId:{},onPartitionsRevoked:{}", clusterId, v.toString());
+            LOGGER.debug("clusterId:{},onPartitionsRevoked:{}, position is {}",
+                    clusterId, v.toString(), consumer.position(v));
         });
-        if (Objects.nonNull(ackOffsetMap) && Objects.nonNull(commitOffsetMap)) {
-            ackRevokedPartitions(collection);
+
+        try {
+            if (Objects.nonNull(ackOffsetMap) && Objects.nonNull(commitOffsetMap)) {
+                //sleep 10s to wait un ack messages
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java:93]: Add one blank after `//`.

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java [hunk 4]
```diff
@@ -59,9 +59,10 @@ public class SortClientConfig implements Serializable {
     private int maxEmptyPollSleepMs = 500;
     private int emptyPollTimes = 10;
     private int cleanOldConsumerIntervalSec = 60;
+    private int maxConsumerSize = 5;
 
     public SortClientConfig(String sortTaskId, String sortClusterName, InLongTopicChangeListener assignmentsListener,
-            ConsumeStrategy consumeStrategy, String localIp) {
+                            ConsumeStrategy consumeStrategy, String localIp) {
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java:65]: No need for these indents.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java (hunk 4 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/api/SortClientConfig.java]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java (hunk 3 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java (hunk 2 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/AckOffsetOnRebalance.java]: fixed, thx

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/fetcher/kafka/KafkaMultiTopicsFetcher.java]: fixed, thx

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**